packages/flutter_tools/lib/src/linux/build_linux.dart

```diff
@@ -108,6 +107,15 @@ Future<void> buildLinux(
       'dart devtools --appSizeBase=$relativeAppSizePath'
     );
   }
+
+  final String? binaryName = getCmakeExecutableName(linuxProject);
+  final File appFile = buildDirectory
+    .childDirectory('bundle')
+    .childFile('$binaryName');
```

Comment:
Does this work?

```suggestion
    .childFile(binaryName);
```

Reply:
Can be null, so no. The check just below handles this case